### PR TITLE
[JSONWebService Refactor] LPD-27115 Remove JSONWebServiceActionManager public apis to register JSON web services, publish using OSGI component with JSONWebService annotation instead

### PR DESCRIPTION
**Pull request by julschong**
Resubmitted from: https://github.com/liferay-core-infra/liferay-portal/pull/7089

This PR removes  3 methods from JSONWebServiceActionsManger:
1. registerService() 
2. registerJSONWebServiceAction() and unregisterJSONWebServiceActions()

Example of using OSGI component way is already giving in : com.liferay.portal.remote.json.web.service.web.client.test.JSONWebServiceTrackerTest

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:core-functional

**Comment by julschong** [comment]
ci:report:2573952822

**Comment by julschong** [comment]
core-functional unique failure
```
Unique Failures			
LocalFile.Clustering#EnableAndDisableFeatureFlagOnMasterNode	https://testray.liferay.com/home/-/testray/case_results/2573959196/history	Unique	java.lang.Exception:  ## ## 8 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
LocalFile.Clustering#ValidatePublishOnTwoNodes	https://testray.liferay.com/home/-/testray/case_results/2573959145/history	Unique	java.lang.Exception:  ## ## 6 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
LocalFile.Clustering#ValidateDocumentOnSeparateNodes	https://testray.liferay.com/home/-/testray/case_results/2573959130/history	Unique	java.lang.Exception:  ## ## 6 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
LocalFile.Clustering#CanUpdateLogLevelsForAllNodesFromSlave	https://testray.liferay.com/home/-/testray/case_results/2573958869/history	Unique	java.lang.Exception:  ## ## 6 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
LocalFile.Clustering#ValidateLiveUsers	https://testray.liferay.com/home/-/testray/case_results/2573958856/history	Unique	java.lang.Exception:  ## ## 4 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
LocalFile.Clustering#CanAddCategoryToDocumentOnSlaveNode	https://testray.liferay.com/home/-/testray/case_results/2573959116/history	Unique	java.lang.Exception:  ## ## 4 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
LocalFile.Clustering#ValidateSchedulerOnSeparateNodes	https://testray.liferay.com/home/-/testray/case_results/2573958797/history	Unique	java.lang.Exception:  ## ## 2 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
LocalFile.Clustering#AddAndDeleteBlogEntriesOnSeparateNodes	https://testray.liferay.com/home/-/testray/case_results/2573959088/history	Unique	java.lang.Exception:  ## ## 2 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
LocalFile.Clustering#EnableAndDisableFeatureFlagOnSlaveNode	https://testray.liferay.com/home/-/testray/case_results/2573959209/history	Unique	java.lang.Exception:  ## ## 10 Liferay Exceptions were thrown ##  LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122] LIFERAY_ERROR: com.liferay.portal.kernel.dao.orm.ORMException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#20122]
```

**Comment by julschong** [comment]
rerun relevant

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:report:11318633

**Comment by julschong** [comment]
@tinatian no regression

File: modules/apps/portal-remote/portal-remote-json-web-service-web/src/test/java/com/liferay/portal/remote/json/web/service/web/internal/BaseJSONWebServiceTestCase.java [hunk 1]
```diff
@@ -97,7 +98,12 @@ protected static void registerActionClass(
 			String method = JSONWebServiceMappingResolverUtil.resolveHttpMethod(
 				actionMethod);
```
Comment (julschong) [inline modules/apps/portal-remote/portal-remote-json-web-service-web/src/test/java/com/liferay/portal/remote/json/web/service/web/internal/BaseJSONWebServiceTestCase.java:100]: pushed change to remove the reflection invocation of _ensureOpen()

**Comment by julschong** [comment]
ci:test:core-functional

**Comment by tinatian** [comment]
@julschong I am closing this as JSONWebService is not in our scope, we won't spend time on it

**Comment by julschong** [comment]
@tinatian I closed the related tickets

**Pull request closed without merging**